### PR TITLE
Add `backbone_lr_ratio` for RT-DETR trainer

### PR DESCRIPTION
**Pull request by artest08**
Adds `backbone_lr_ratio` so RT-DETR users can fine-tune a pretrained backbone at a lower learning rate than the detection head — a common recipe when adapting a pretrained model to a new dataset (`0.1` is typical). The argument is scoped to `RTDETRTrainer.build_optimizer` so YOLO training is untouched, and the default `1.0` short-circuits to the base trainer so existing RT-DETR runs and their saved optimizer state remain bit-identical.

```python
from ultralytics import RTDETR

RTDETR("rtdetr-l.pt").train(data="coco8.yaml", epochs=100, imgsz=640, backbone_lr_ratio=0.1)
```

Backbone layers are identified from `model.yaml["backbone"]` length via `unwrap_model(model)`; when the ratio is below `1.0`, each parameter group (weight / bn / bias / muon) is split into head and backbone subgroups, and the backbone subgroup's learning rate is set to `lr0 * backbone_lr_ratio`. MuSGD, AdamW, and SGD optimizer paths are all preserved, including the named-dict parameter-group layout and `param_group` labels.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds RT-DETR support for a new `backbone_lr_ratio` training argument, enabling a lower learning rate for backbone layers while keeping the detection head at the base learning rate. 🚀

### 📊 Key Changes
- Added `backbone_lr_ratio` to `default.yaml` with a default value of `1.0`.
- Implemented a custom `build_optimizer()` in `ultralytics/models/rtdetr/train.py` to split RT-DETR optimizer parameter groups into backbone and head groups with separate learning rates.
- Preserved existing behavior when `backbone_lr_ratio=1.0` by falling back to the parent optimizer builder.
- Added validation to reject invalid `backbone_lr_ratio` values less than or equal to zero.
- Included support across optimizer types, including `auto` selection and `MuSGD`, with logging for head/backbone parameter group breakdowns. ⚙️
- Updated RT-DETR docs with a practical training example showing how to use `backbone_lr_ratio`.
- Added the new argument to the shared training arguments macro and clearly documented that it is currently only honored by the RT-DETR trainer. 📚

### 🎯 Purpose & Impact
- Improves fine-tuning control for RT-DETR, especially when using a pretrained backbone with a newly initialized detection head.
- Helps users train backbone layers more conservatively, which can improve stability during transfer learning and reduce over-updating pretrained features.
- Keeps backward compatibility intact for existing RT-DETR workflows by default.
- Enhances discoverability through both model-specific and shared training documentation, reducing confusion about where the argument applies. ✅

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @artest08, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness; an engineer will assist you soon. Please review the checklist below to help ensure a smooth integration. 😊

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/models/rtdetr/train.py [hunk 1]
```diff
@@ -83,6 +87,105 @@ def build_dataset(self, img_path: str, mode: str = "val", batch: int | None = No
...
+            LOGGER.info(
+                f"{colorstr('optimizer:')} 'optimizer=auto' found, "
+                f"ignoring 'lr0={self.args.lr0}' and 'momentum={self.args.momentum}' and "
+                f"determining best 'lr0' and 'momentum' automatically for AdamW... "
+            )
+            nc = self.data.get("nc", 10)
+            name, lr, momentum = "AdamW", round(0.002 * 5 / (4 + nc), 6), 0.9
+            self.args.warmup_bias_lr = 0.0
+
+        use_muon = name == "MuSGD"
```
Comment (UltralyticsAssistant) [inline ultralytics/models/rtdetr/train.py:125]: ⚠️ **HIGH**: `name` is checked for `"auto"` and `"MuSGD"` before it is normalized to the canonical casing below. That makes case-insensitive values behave incorrectly: `optimizer=Auto` skips the auto branch and later falls through to `NotImplementedError`, while `optimizer=musgd` leaves `use_muon=False` and eventually calls `getattr(torch.optim, "MuSGD")`, which will fail. Normalize `name` once before any special-case checks so these accepted aliases keep working.

File: ultralytics/models/rtdetr/train.py [hunk 2]
```diff
@@ -83,6 +87,105 @@ def build_dataset(self, img_path: str, mode: str = "val", batch: int | None = No
...
+        elif name in {"SGD", "MuSGD"}:
+            optim_args = dict(lr=lr, momentum=momentum, nesterov=True)
+        else:
+            raise NotImplementedError(
+                f"Optimizer '{name}' not found in list of available optimizers {optimizers}. "
+                "Request support for addition optimizers at https://github.com/ultralytics/ultralytics."
+            )
+
+        backbone_lr = lr * backbone_lr_ratio
+        head_groups = [
```
Comment (UltralyticsAssistant) [inline ultralytics/models/rtdetr/train.py:159]: ⚠️ **HIGH**: This changes the param-group ordering from the parent trainer. Ultralytics warmup logic treats `optimizer.param_groups[0]` as the bias group when applying `warmup_bias_lr`; here group 0 is `head_weight`, so enabling `backbone_lr_ratio != 1` will warm up decayed weights as if they were biases and leave the actual bias groups on the normal schedule. Please preserve the parent ordering (bias first) or update the warmup path to target groups by metadata instead of index.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Nice feature addition and the RT-DETR-specific fallback is a good way to avoid changing default behavior, but there are two training regressions to address before merge: optimizer-name handling is unintentionally case-sensitive in the new path, and the custom param-group order breaks the existing bias warmup assumption.

💬 Posted 2 inline comments
